src/piral-build.test.ts

```diff
@@ -7,63 +7,180 @@ runTests('piral-build', ({ test, setup }) => {
+    async (ctx) => {
+      try {
+        await ctx.run(`npx piral build --type some-invalid-type-standard-template`);
+        expect(true).toBe(false);
+      } catch {}
+
+      await ctx.assertFiles({
+        'dist/release': false,
+        'dist/emulator': false,
+      });
+    },
+  );
+
   test(
-    'release-standard-template',
-    'can produce a release build',
+    'release-standard-template-with-default-public-url',
+    'can produce a release build with default public url',
     ['build.piral'],
     async (ctx) => {
-      await ctx.run(`npx piral build --type release`);
+      await ctx.run(`npx piral build --type release --public-url '/'`);
 
       await ctx.assertFiles({
         'dist/release/index.html'(content: string) {
           expect(content).not.toBe('');
+          expect(content).toContain('<script defer="defer" src="/index.');
         },
       });
     },
   );
 
   test(
-    'emulator-standard-template',
-    'can produce an emulator build',
+    'release-standard-template-with-a-different-public-url',
+    'can produce a release build with a different public url ./',
     ['build.piral'],
     async (ctx) => {
-      await ctx.setFiles({
-        'package.json'(content: string) {
-          const packageJson = JSON.parse(content);
-          packageJson.name = 'app-shell';
-          return JSON.stringify(packageJson, undefined, 2);
+      await ctx.run(`npx piral build --type release --public-url 'different-public-url/'`);
+
+      await ctx.assertFiles({
+        'dist/release/index.html'(content: string) {
+          expect(content).not.toBe('');
+          expect(content).toContain('src="/different-public-url/');
+          expect(content).toContain('<link href="/different-public-url/');
         },
       });
+    },
+  );
 
-      await ctx.run(`npx piral build --type emulator`);
+  test(
+    'release-standard-template-with-default-target',
+    'can produce a release build with default target ./dist',
+    ['build.piral'],
+    async (ctx) => {
+      await ctx.run(`npx piral build --type release --target`);
 
       await ctx.assertFiles({
-        'dist/emulator/*.tgz'(files: Array<string>) {
-          expect(files).toHaveLength(1);
-          ctx.setRef('emulator', files[0]);
+        'dist/release/index.html'(content: string) {
+          expect(content).not.toBe('');
         },
       });
     },
   );
 
   test(
-    'emulator-scaffold',
-    'can scaffold from the emulator',
-    ['#empty'],
+    'release-standard-template-with-a-different-target',
+    'can produce a release build with a different target',
+    ['build.piral'],
     async (ctx) => {
-      const source = ctx.getRef('emulator');
+      await ctx.run(`npx piral build --type release --target './different-target'`);
 
-      await ctx.run(npmInit(`pilet@${cliVersion}`, `--source ${source} --bundler none --defaults`));
+      await ctx.assertFiles({
+        'different-target/release/index.html'(content: string) {
```

Comment:
Great, but since we use `--type release` we can still check that there is no `different-target/emulator`.